### PR TITLE
Jessica/9125/inject provenance resource when missing

### LINKED ISSUE #9125: Inject Provenance resource when missing
### User Story:
As a Connectathon attender, I want to ensure that people can send me messages that do not contain a provenance without erroring out. In the future, as a Reportstream sender, I want to be able to successfully send messages without a provenance. 

### Description/Use Case
We currently produce an error when a message is sent without a provenance. Instead of throwing an error, we want to inject the provenance message so that the error is not produced. 

### Acceptance Criteria
- [x] A message can be sent without the provenance and it will not fail.

### PR DESCRIPTION
**Pull request by JessicaWNava**
This PR adds a provenance resource when one is missing as well as sets the processing mode filter to evaluate to true so that the sample message for Connectathon can be processed as is. 

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. Navigate to http://build.fhir.org/ig/HL7/home-lab-report/Bundle-bundle-at-home-in-vitro-test-results-p-noelle.json.html to get the test message
2. Update the state in the message to be IG
3. Send the message through /reports
4. Check that the message is sent to an organization via /waters/report/<insert_id_here>/history

## Changes
- Add the provenance resource when it is missing from the message
- Set the processing mode filter to something that will make it evaluate to true. 

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #9125 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

**Comment by JessicaWNava** [comment]
@arnejduranovic and @jimmyfagan just need a re review so these changes will actually go to master this time instead of the connectathon branch

**Review by jimmyfagan — APPROVED** [review]
Still looks good!

**Review by arnejduranovic — APPROVED** [review]

File: prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt [hunk 1]
```diff
@@ -79,7 +80,8 @@ object FHIRBundleHelpers {
         val provenanceResource = try {
             fhirBundle.entry.first { it.resource.resourceType.name == "Provenance" }.resource as Provenance
         } catch (e: NoSuchElementException) {
-            throw IllegalStateException("The FHIR bundle does not contain a Provenance resource")
+            fhirBundle.addEntry().resource = Provenance()
+            fhirBundle.entry.first { it.resource.resourceType.name == "Provenance" }.resource as Provenance
```
Comment (jimmyfagan) [inline prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt:84]: Looks like this change breaks test `manipulating bundle without provenance`, that might not have existed in the other branch. Just make sure you can run `test` and `testIntegration` locally to make sure there aren't other cases we missed here.

**Review by jimmyfagan — CHANGES_REQUESTED** [review]
The CI checks failed because some of the tests are breaking from the changes here, the failing tests can probably just be updated and we'll be okay.

**Review by jimmyfagan — APPROVED** [review]
Looks good!

**Pull request merged**